### ISSUE TITLE
Bazel integration tests loop forever writing log files on failure

### ISSUE THREAD

**Issue by jmillikin-stripe** [bazelbuild/bazel#4085]
Discovered while testing https://github.com/bazelbuild/bazel/issues/4028 -- this might apply only to `cpp_darwin_integration_test.sh`, I didn't explore too deep.

Looking at `test_osx_cc_wrapper_rpaths_handling()` as an example:

```bash
74   assert_build //cpp/rpaths:test >& $TEST_log || fail "//cpp/rpaths:test didn't build"
75   # Paths originally hardcoded in the binary assume workspace directory. Let's change the
76   # directory and execute the binary to test whether the paths in the binary have been
77   # updated to use @loader_path.
78   cd bazel-bin
79   ./cpp/rpaths/test >& $TEST_log || \
80       fail "//cpp/rpaths:test execution failed, expected to return 0, but got $?"
81 }
```

It's intended to try building, with logs sent to a file, and fail if the build failed. Straightforward. However, in `testenv.sh`:

```
515 function assert_build() {
516   bazel build -s --verbose_failures $* || fail "Failed to build $*"
517 }
```

The extra layer of `fail`, plus the shell output redirect, seems to send Bash into an infinite loop where it just writes the same log message to `$TEST_log` over and over until the disk is full.

**Comment by mhlopko** [comment]
Will be fixed by #4029

**Comment by jmillikin-stripe** [comment]
This is fixed now.
